test/example/generated/ExampleDialect.cpp.inc

```diff
@@ -138,7 +140,14 @@ attrBuilder.addAttribute(::llvm::Attribute::NoUnwind);
 attrBuilder.addMemoryAttr(::llvm::MemoryEffects(::llvm::MemoryEffects::Location::InaccessibleMem, ::llvm::ModRefInfo::ModRef));
 m_attributeLists[3] = ::llvm::AttributeList::get(context, ::llvm::AttributeList::FunctionIndex, attrBuilder);
 }
-}
+
+    static const ::llvm::TargetExtTypeClass classXdVectorType(
+          "xd.vector", false,
+          [](const ::llvm::TargetExtType *type) {
+            return ::llvm::cast<XdVectorType>(type)->getTypeInfo();
+          });
+    context.registerTargetExtTypeClass(&classXdVectorType);
```

Comment:
Shouldn't we only register upon the first pass?

Currently we re-create a dialect every time a DialectContext is created for it (not sure whether that's necessary), so we'd register the same class multiple times.

Reply:
The registration is per-LLVMContext, and we should really only be creating a DialectContext once per LLVMContext.

I suppose that this wasn't a strict rule previously, and now it becomes one, but I think it's a reasonable rule to have.